### ISSUE TITLE
Test failure: testStringUTF8Encoding (google.protobuf.internal.reflection_test.ReflectionTest)

### ISSUE THREAD

**Issue by dmoisset** [machinalis/protobuf-python3#45]
```
======================================================================
FAIL: testStringUTF8Encoding (google.protobuf.internal.reflection_test.ReflectionTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dmoisset/canonical/protobuf/python/google/protobuf/internal/reflection_test.py", line 1355, in testStringUTF8Encoding
    setattr, proto, 'optional_bytes', 'unicode object')
AssertionError: TypeError not raised by setattr

```

**Comment by juannorris** [comment]
After changes made to reflection_test.py still getting a failure. Seems like implementation needs some changes (this is working in 2.6).

```
======================================================================
FAIL: testStringUTF8Encoding (google.protobuf.internal.reflection_test.ReflectionTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jnorris/projects/3k-protobuf/python/google/protobuf/internal/reflection_test.py", line 1365, in testStringUTF8Encoding
    self.assertEqual(proto.optional_string, 'Testing')
AssertionError: b'Testing' != 'Testing'
```

**Comment by juannorris** [comment]
This is how code was before changes (where test is failling).
I think the 'porting' is correct:

```
# Assign a value of type 'str' which can be encoded in UTF-8.
    proto.optional_string = str('Testing')
    self.assertEqual(proto.optional_string, unicode('Testing'))
```

str('Testing')         ==>   b'Testing'
unicode('Testing') ==>   'Testing'

**Comment by dmoisset** [comment]
Fixed in d57d8ed.

This test used strongly that in python 2.6 , b'x' == u'x'. The test itself had to be made more strict to make sense
